src/Strategy/UserSpecified/YamlDependencies.hs

```diff
@@ -0,0 +1,120 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Strategy.UserSpecified.YamlDependencies
+  ( analyze,
+    discover,
+    buildGraph,
+    UserDependencies (..),
+  )
+where
+
+import Control.Effect.Diagnostics
+import Data.Aeson
+import Data.Aeson.Types (Parser)
+import qualified Data.Map.Strict as M
+import Data.Text (Text, unpack)
+import Debug.Trace
+import DepTypes
+import Discovery.Walk
+import Effect.ReadFS
+import Graphing (Graphing)
+import qualified Graphing
+import Path
+import Types
+
+discover :: (Has ReadFS sig m, Has Diagnostics sig m, Has ReadFS rsig run, Has Diagnostics rsig run) => Path Abs Dir -> m [DiscoveredProject run]
+discover dir = map mkProject <$> findProjects dir
+
+-- Only search for fossa-deps.yaml in the root directory. We can extend this to subdirectories in the future.
+findProjects :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs Dir -> m [UserDependenciesYamlProject]
```

Comment:
Given that we only want to find the file in the root, we can skip the walk and simplify this to something like:
```hs
findProjects dir = do
  exists <- doesFileExist $(mkRelFile ".fossa.yml")
  if exists
    then .. -- make the project
    else pure []
```

Reply:
For some reason, I kept thinking that `mkRelFile` was actually creating a file at that location, thanks.